src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -783,6 +783,9 @@ bool emberAfOperationalCredentialsClusterUpdateNOCCallback(app::CommandHandler *
     err = fabric->SetICACert(ICACValue);
     VerifyOrExit(err == CHIP_NO_ERROR, nocResponse = ConvertToNOCResponseStatus(err));
 
+    err = fabric->ValidateNewNOCCredentials();
```

Comment:
No, this is not right.  We've already updated our fabric with the new (possibly-invalid) NOC.  What's the point of validating it now?

What might make more sense is:

1. Set the new state on `gFabricBeingCommissioned`, not our existing fabric info.  Might involve setting some state on it (e.g. root cert, vendor id, label) from our existing fabric.
2. Call SetFabricInfo with that.  This will do the validation as needed on the incoming fabric info.
3. In SetFabricInfo, check that either mFabric is undefined or the incoming fabric id matches (to allow both the AddNOC and UpdateNOC cases).

Or something along those lines.

Reply:
OK, I have updated the implementation and ValidateNewNOCCredentials() will take input parameters and generate some outputs.